sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -245,6 +246,28 @@ object DataSourceV2Strategy extends Strategy with PredicateHelper {
       }.toArray
       DeleteFromTableExec(r.table.asDeletable, filters) :: Nil
 
+    case UpdateTable(r: DataSourceV2Relation, attrs, values, condition) =>
+      val nested = attrs.asInstanceOf[Seq[Any]].filterNot(_.isInstanceOf[AttributeReference])
```

Comment:
Can we remove `.asInstanceOf[Seq[Any]]` now?

Reply:
No. As I explained above, `Seq[NamedExpression]` can not resolve the problem. See https://github.com/apache/spark/pull/25626#discussion_r320605605.